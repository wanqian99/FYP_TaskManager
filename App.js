import 'react-native-gesture-handler';

import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View } from 'react-native';
import { SafeAreaProvider, SafeAreaView } from 'react-native-safe-area-context';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

import NavigationDrawerStructure from "./components/customDrawer.js"
import LoginScreen from './screens/loginScreen.js';
import RegistrationScreen from './screens/registrationScreen.js';

import { LogBox } from 'react-native';
LogBox.ignoreLogs(['Warning: ...']); // Ignore log notification by message
LogBox.ignoreAllLogs();//Ignore all log notifications

const Stack = createNativeStackNavigator();

export default function App() {
  return (
    <SafeAreaProvider>
        <NavigationContainer>
            <Stack.Navigator 
                screenOptions={({ navigation }) => ({
                    headerShown: false,
                })}
            >
                <Stack.Screen name="Login" component={LoginScreen} />
                <Stack.Screen name="Register" component={RegistrationScreen} />
                <Stack.Screen name="Main" component={NavigationDrawerStructure} />
            </Stack.Navigator>
        </NavigationContainer>
    </SafeAreaProvider>
  );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
});
