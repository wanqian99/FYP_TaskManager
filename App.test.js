import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import App from './App.js';

describe("<App />", () => {
    // it("has 1 child", () => {
    //     const tree = renderer.create(<App />).toJSON();
    //     expect(tree.children.length).toBe(1);
    // });

    it("renders correctly", () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // it("renders Hello World message on the home page", async () => {
    //     render(<App />);
    //     expect(screen.getByText("Hello, World!")).toBeDefined()
    // });
});