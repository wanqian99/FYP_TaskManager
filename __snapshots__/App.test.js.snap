// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
